clcache.py

```diff
@@ -793,13 +831,17 @@ def _outputFileFromArgument(options, argumentName, sourceFile, extension):
     @staticmethod
     def _parseOptionsAndFiles(cmdline):
         optionsWithParameter = {
-            'Ob', 'Gs', 'Fa', 'Fd', 'Fm',
-            'Fp', 'FR', 'doc', 'FA', 'Fe',
-            'Fo', 'Fr', 'AI', 'FI', 'FU',
-            'D', 'U', 'I', 'Zp',
-            'F', 'Fi', 'MP', 'Tc', 'Tp', 'V',
-            'w1', 'w2', 'w3', 'w4', 'wd', 'we', 'wo', 'Wv',
-            'Yc', 'Yl', 'Yu', 'Zm',
+            ArgumentT1('FI'), ArgumentT1('U'), ArgumentT1('I'),
+            ArgumentT2('F'), ArgumentT2('FU'),
+            ArgumentT3('doc'), ArgumentT3('FA'), ArgumentT3('FR'), ArgumentT3('Fr'),
+            ArgumentT3('Gs'), ArgumentT3('MP'), ArgumentT3('Wv'), ArgumentT3('Yc'),
+            ArgumentT3('Yu'), ArgumentT3('Zp'),
+            ArgumentT4('AI'), ArgumentT4('D'), ArgumentT4('Fa'), ArgumentT4('Fd'),
+            ArgumentT4('Fe'), ArgumentT4('Fi'), ArgumentT4('Fm'), ArgumentT4('Fo'),
+            ArgumentT4('Fp'), ArgumentT4('Ob'), ArgumentT4('Tc'), ArgumentT4('Tp'),
+            ArgumentT4('V'), ArgumentT4('w1'), ArgumentT4('w2'), ArgumentT4('w3'),
+            ArgumentT4('w4'), ArgumentT4('wd'), ArgumentT4('we'), ArgumentT4('wo'),
+            ArgumentT4('Yl'), ArgumentT4('Zm'),
```

Comment:
Are you sure this classification is correct? E.g. `/w3 1234` is not equivalent to `/w31234`?